### ISSUE TITLE
[Bug] kube-vip svc_election: false is misbehaving in table-mode.

### ISSUE THREAD

**Issue by Cellebyte** [kube-vip/kube-vip#627]
**Describe the bug**
We currently use kube-vip without `svc_election` in table mode to allow ECMP for our VIPs.
When pod terminating a LoadBalancer service is moved to a different node.
When this happens the ip is still advertised from the first node and additionally from the new node.
This unfortunately causes issues in routing as one endpoint accepts traffic and the other one does not.

**To Reproduce**
1. Install kube-vip in table mode and without `svc_election: true`
2. Create a loadbalancer service which selects a single pod.
3. Move that pod from one server to another and investigate the routing table.

**Expected behavior**
The endpoints should be cleaned up when they move to different nodes.
Additionally kube-vip updates the Service ressource all the time when there was no service change.

**Screenshots**
```bash
kubectl get services -n ingress-system
NAME                                                  TYPE           CLUSTER-IP                EXTERNAL-IP      PORT(S)                         AGE
management-nginx-ingress-controller                   LoadBalancer   172.30.114.2              10.0.0.1    80:31189/TCP,443:31567/TCP      22h
management-nginx-ingress-controller-default-backend   ClusterIP      172.30.71.254             <none>           80/TCP                          22h
management-nginx-ingress-controller-legacy            LoadBalancer   172.30.71.239             10.0.0.1    8080:31317/TCP,8443:30173/TCP   22h
management-nginx-ingress-controller-metrics           ClusterIP      172.30.100.124            <none>           9913/TCP                        22h
```
```bash
kubectl get endpoints -n ingress-system
NAME                                                  ENDPOINTS                                                      AGE
management-nginx-ingress-controller                   172.30.0.198:443,172.30.1.33:443,172.30.1.35:443 + 3 more...   22h
management-nginx-ingress-controller-default-backend   172.30.1.34:8080                                               22h
management-nginx-ingress-controller-legacy            172.30.0.198:443,172.30.1.33:443,172.30.1.35:443 + 3 more...   22h
management-nginx-ingress-controller-metrics           172.30.0.198:10254,172.30.1.33:10254,172.30.1.35:10254         22h
```

```bash
kubectl get pods -A -o wide -l app.kubernetes.io/instance=management
NAMESPACE        NAME                                                              READY   STATUS    RESTARTS   AGE   IP             NODE                        NOMINATED NODE   READINESS GATES
ingress-system   management-nginx-ingress-controller-5b594857b9-jx5jx              1/1     Running   0          22h   172.30.1.33    ship-lab-1-md-gxntn-sqk4n   <none>           <none>
ingress-system   management-nginx-ingress-controller-5b594857b9-ms479              1/1     Running   0          22h   172.30.1.35    ship-lab-1-md-gxntn-sqk4n   <none>           <none>
ingress-system   management-nginx-ingress-controller-5b594857b9-vxd6g              1/1     Running   0          22h   172.30.0.198   ship-lab-1-md-gxntn-z5c8f   <none>           <none>
ingress-system   management-nginx-ingress-controller-default-backend-6976bdvp9fl   1/1     Running   0          22h   172.30.1.34    ship-lab-1-md-gxntn-sqk4n   <none>           <none>
```

```bash
kubectl get nodes
NAME                        STATUS   ROLES           AGE     VERSION
ship-lab-1-gn9xw            Ready    control-plane   2d19h   v1.26.9
ship-lab-1-md-gxntn-sqk4n   Ready    worker          2d19h   v1.26.9
ship-lab-1-md-gxntn-z5c8f   Ready    worker          2d19h   v1.26.9
ship-lab-1-s2ks9            Ready    control-plane   2d18h   v1.26.9
ship-lab-1-srjj2            Ready    control-plane   2d18h   v1.26.9
```

```bash
root@ship-lab-1-s2ks9:/home/user# ip r show table all | grep 10.0.0.1
10.0.0.1 nhid 146 via inet6 fe80::c0a0:59ff:fe58:3f23 dev <vrf_interface> table 3 proto bgp metric 20
10.0.0.1 dev lo table 198 # this should not be here.
10.0.0.1 nhid 448 dev lo proto bgp src <node-ip> metric 20
```

**Environment (please complete the following information):**
 - OS/Distro: Ubuntu 20.04
 - Kubernetes Version: v.1.26.9
 - Kube-vip Version: 0.6.2

**`Kube-vip.yaml`:**

```yaml
# Source: schiff-loadbalancing/templates/daemonsets/kube-vip-svc_kube-system.yaml
apiVersion: apps/v1
kind: DaemonSet
metadata:
  creationTimestamp: null
  labels:
    app.kubernetes.io/name: kube-vip-svc
    app.kubernetes.io/version: v0.6.2
  name: kube-vip-svc
  namespace: kube-system
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: kube-vip-svc
  template:
    metadata:
      creationTimestamp: null
      labels:
        app.kubernetes.io/name: kube-vip-svc
        app.kubernetes.io/version: v0.6.2
    spec:
      containers:
      - args:
        - manager
        env:
        - name: vip_loglevel
          value: "10"
        - name: vip_arp
          value: "false"
        - name: port
          value: "6443"
        - name: vip_interface
          value: lo
        - name: vip_cidr
          value: "32"
        - name: svc_enable
          value: "true"
        - name: vip_routingtable
          value: "true"
        - name: vip_address
        - name: prometheus_server
          value: :2112
        - name: svc_election
          value: "false"
        image: ghcr.io/kube-vip/kube-vip:v0.6.2
        imagePullPolicy: Always
        name: kube-vip
        resources: {}
        securityContext:
          capabilities:
            add:
            - NET_ADMIN
            - NET_RAW
      hostNetwork: true
      serviceAccountName: kube-vip
      nodeSelector:
        node-role.kubernetes.io/worker: ""
  updateStrategy: {}
```

**Additional context**

**Comment by thebsdbox** [comment]
I'll need to update the e2e tests to have better visibility into this, but I suspect someone has broken some of the code that ensures watchers/local sessions are garbage collected. Hopefully will be a quick fix.

**Comment by Cellebyte** [comment]
Yeah I will try to replicate the issue tomorrow and provide some screenshots.

**Comment by thebsdbox** [comment]
This should now be fixed, I'll roll a new release shortly!

**Comment by Cellebyte** [comment]
Yep it was fixed with release v0.7.0
